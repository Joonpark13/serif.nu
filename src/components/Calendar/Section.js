import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import Paper from '@material-ui/core/Paper';

export const styles = {
  paper: {
    position: 'absolute',
    top: ({ hour, section }) => {
      const schedule = section.schedule.find(time => time.start.hour === hour);
      /* eslint-disable-next-line prefer-destructuring */
      const minute = schedule.start.minute;
      const offset = Math.round(minute / 60 * 100);
      return `${offset}%`;
    },
  },
};

function Section({ classes, section }) {
  return (
    <Paper className={classes.paper}>
      {section.course}
    </Paper>
  );
}

Section.propTypes = {
  hour: PropTypes.number.isRequired, // eslint-disable-line react/no-unused-prop-types
  section: PropTypes.objectOf(PropTypes.any).isRequired, // TODO
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export { Section as UnstyledSection };
export default withStyles(styles)(Section);
