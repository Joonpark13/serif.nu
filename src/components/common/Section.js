import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

function Section({ classes, leftHeaderContent, rightHeaderContent, sectionName, onClick }) {
  return (
    <Paper elevation={2} className={classes.paper} onClick={onClick}>
      <div className={classes.container}>
        <div className={classes.header}>
          <Typography variant="caption" className={classes.text}>
            {leftHeaderContent}
          </Typography>

          <Typography variant="caption" className={classes.text}>
            {rightHeaderContent}
          </Typography>
        </div>

        <Typography variant="subtitle2" className={`${classes.text} ${classes.name}`} noWrap>
          {sectionName}
        </Typography>
      </div>
    </Paper>
  );
}

Section.propTypes = {
  classes: PropTypes.shape({
    paper: PropTypes.string,
    container: PropTypes.string,
    header: PropTypes.string,
    text: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
  leftHeaderContent: PropTypes.string,
  rightHeaderContent: PropTypes.string,
  sectionName: PropTypes.string,
  onClick: PropTypes.func,
};

Section.defaultProps = {
  leftHeaderContent: '',
  rightHeaderContent: '',
  sectionName: '',
  onClick: () => {},
};

export default Section;
