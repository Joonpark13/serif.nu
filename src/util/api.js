/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
