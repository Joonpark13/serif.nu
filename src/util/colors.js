export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
