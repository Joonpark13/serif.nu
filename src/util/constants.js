/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
