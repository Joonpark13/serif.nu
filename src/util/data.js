import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
