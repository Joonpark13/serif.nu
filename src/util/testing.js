/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
