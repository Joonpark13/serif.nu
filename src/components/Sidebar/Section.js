import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import { getFormattedClassSchedule } from 'util/time';

export const styles = {
  sectionTitle: {
    fontWeight: 'bold',
  },
};

function Section({ addSection, section, disabled, classes }) {
  return (
    <ListItem
      key={section.id}
      button
      onClick={() => addSection(section)}
      disabled={disabled}
    >
      <ListItemText>
        <Typography variant="h6" className={classes.sectionTitle}>
          {`Section ${section.sectionNumber}`}
        </Typography>

        {section.topic && <Typography>{section.topic}</Typography>}

        {section.schedule.map((scheduleObj, index) => (
          /* eslint-disable react/no-array-index-key */
          <Typography key={index}>
            {getFormattedClassSchedule(scheduleObj)}
          </Typography>
          /* eslint-enable react/no-array-index-key */
        ))}

        <Typography>{section.schedule[0].location}</Typography>

        <Typography>
          {section.instructors.map((teacher, idx) => (
            `${teacher}${idx !== section.instructors.length - 1 ? ', ' : ''}`))
          }
        </Typography>
      </ListItemText>
    </ListItem>
  );
}

Section.propTypes = {
  addSection: PropTypes.func.isRequired,
  section: PropTypes.objectOf(PropTypes.any).isRequired, // TODO
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  disabled: PropTypes.bool,
};

Section.defaultProps = {
  disabled: false,
};

export { Section as UnstyledSection };
export default withStyles(styles)(Section);
