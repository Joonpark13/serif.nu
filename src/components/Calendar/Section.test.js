import React from 'react';
import { shallow } from 'enzyme';
import { mockStyles } from 'util/testing';
import { UnstyledSection, styles } from './Section';

describe('Section', () => {
  describe('dynamic styles', () => {
    const hour = 10;
    const section = {
      schedule: [{
        dow: ['Mo'],
        start: {
          hour: 10,
          minute: 30,
        },
        end: {
          hour: 11,
          minute: 50,
        },
      }],
    };

    expect(styles.paper.top({ hour, section })).toBe('50%');
  });

  it('renders correctly', () => {
    const classes = mockStyles(styles);
    const testSection = { id: 12345, course: '101-1' };
    const wrapper = shallow(
      <UnstyledSection hour={10} section={testSection} classes={classes} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
