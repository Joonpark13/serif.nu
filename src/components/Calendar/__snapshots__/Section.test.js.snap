// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    classes={
      Object {
        "paper": "",
      }
    }
    hour={10}
    section={
      Object {
        "course": "101-1",
        "id": 12345,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "101-1",
      "className": "",
    },
    "ref": null,
    "rendered": "101-1",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "101-1",
        "className": "",
      },
      "ref": null,
      "rendered": "101-1",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
